server/src/main/java/io/crate/planner/operators/Collect.java

```diff
@@ -334,7 +334,11 @@ public LogicalPlan replaceSources(List<LogicalPlan> sources) {
     public LogicalPlan pruneOutputsExcept(SequencedCollection<Symbol> outputsToKeep) {
         LinkedHashSet<Symbol> newOutputs = new LinkedHashSet<>();
         for (Symbol outputToKeep : outputsToKeep) {
-            SymbolVisitors.intersection(outputToKeep, outputs, newOutputs::add);
+            SymbolVisitors.intersection(outputToKeep, outputs, needle -> {
```

Comment:
Why can't we use the symbol from `outputToKeep` ? If this logic is needed to make the pruning for `Collect` work correcly, then i think this is the wrong place to add it.  `SymbolVisitors` should do this.

Reply:
Would this work if we looped over the outputs instead?

```java
 for (Symbol output : outputs) {
    SymbolVisitors.intersection(output, outputsToKeep, newOutputs::add);
    ...

```

Reply:
>Would this work if we looped over the outputs instead?

I tried this. It works for this particular regression but makes CASE query in https://github.com/crate/crate/issues/16047 fail.

Outputs in the query from https://github.com/crate/crate/issues/16047 are just references and they don't need to be handles by a visitor, it's `outputsToKeep` has a Function and needs some extra traversing.

Reply:
> then i think this is the wrong place to add it. SymbolVisitors should do this.

I though about introducing a dedicated intersection but since it's a single usage I thikn it's not worth it. 

If you mean we need to change `SymbolVisitors.intersection` to take items from the haystack: I'm not sure we should do it for a hotfix. Surface area where we use `intersection` is too big and this is narrow fix for a concrete regression in a concrete Operator.